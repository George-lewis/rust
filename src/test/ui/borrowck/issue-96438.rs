use std::cell::RefCell;

fn foo(cell: &RefCell<Option<Vec<()>>>) {
    cell.borrow_mut().unwrap().pop().unwrap();
    //~^ ERROR cannot move out of dereference of `RefMut<'_, Option<Vec<()>>>` [E0507]
    //~| NOTE move occurs because value has type `Option<Vec<()>>`, which does not implement the `Copy` trait
    //~| HELP consider borrowing the `Option`'s content
}

fn main() {
    let cell = RefCell::new(None);
    foo(&cell);
}