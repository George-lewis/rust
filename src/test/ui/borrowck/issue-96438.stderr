error[E0507]: cannot move out of dereference of `RefMut<'_, Option<Vec<()>>>`
  --> $DIR/issue-96438.rs:4:5
   |
LL |     cell.borrow_mut().unwrap().pop().unwrap();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ move occurs because value has type `Option<Vec<()>>`, which does not implement the `Copy` trait
   |
help: consider borrowing the `Option`'s content
   |
LL |     cell.borrow_mut().as_mut().unwrap().pop().unwrap();
   |                      +++++++++
LL |     cell.borrow_mut().as_ref().unwrap().pop().unwrap();
   |                      +++++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0507`.
